### PR TITLE
Fully Support Agents File & Web Retrieval

### PR DESCRIPTION
**Pull request by tmichaeldb**
## Description

This PR adapts the loaders & splitters from [our `email_rag` repository](https://github.com/mindsdb/email_rag/tree/main) into this repository to be used for retrieval. Additionally, we expose a new `PUT` endpoint at `/<project_name>/knowledge_bases/<knowledge_base_name>` that allows our SDK (& users) to add files & websites to a Knowledge Base for later retrieval.

Fixes RAG-3
Fixes RAG-21
Fixes RAG-14

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [x]   Test Locations:
     - `./tests/integrations/utilities/rag/test_file_splitter.py`
     - `./tests/integrations/utilities/rag/test_file_loader.py`
 - [x]   Verification Steps: **New Knowledge Base endpoint tested locally** by making various post requests to a local `mindsdb` instance

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Review by dusvyat — APPROVED** [review]
Nicely done , excellent work 🚀

File: tests/integrations/utilities/rag/data/langchain_readme.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
+---
```
Comment (ea-rus) [inline tests/integrations/utilities/rag/data/langchain_readme.md:1]: Is it just copy of https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/langchain_handler/README.md
If so we could use link to it

File: mindsdb/integrations/utilities/rag/splitters/file_splitter.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+@dataclass
+class FileSplitterConfig:
+    '''Represents configuration needed to split a file into chunks for retrieval.'''
+    # Target chunk size in characters. Not all splitters will adhere exactly to this (it's more of a guideline)
+    chunk_size: int = DEFAULT_CHUNK_SIZE
+    # How many characters each chunk should overlap. Not all splitters will adhere exactly to this (it's more of a guideline)
+    chunk_overlap: int = DEFAULT_CHUNK_OVERLAP
+    # Embeddings to use for semantic chunking (default is OpenAI)
+    embeddings: Embeddings = OpenAIEmbeddings()
```
Comment (ea-rus) [inline mindsdb/integrations/utilities/rag/splitters/file_splitter.py:36]: it requires OPENAI_API_KEY in env

File: mindsdb/api/http/namespaces/knowledge_bases.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+_DEFAULT_MARKDOWN_HEADERS_TO_SPLIT_ON = [
+    ("#", "Header 1"),
+    ("##", "Header 2"),
+    ("###", "Header 3"),
+]
+
+
+def _insert_file_into_knowledge_base(table: KnowledgeBaseTable, file_name: str):
+    file_controller = FileController()
+    splitter = FileSplitter()
```
Comment (ea-rus) [inline mindsdb/api/http/namespaces/knowledge_bases.py:31]: looks like FileSplitter requires 'config' param

**Pull request merged**